framework/src/runtime.rs

```diff
@@ -23,9 +28,11 @@ impl Runtime {
         info!("initializing context:\n{}", configuration);
         let mut context = initialize_system(configuration)?;
         context.start_schedulers();
+        let tokio_rt = TokioRuntime::new()?;
         Ok(Runtime {
             context,
-            on_signal: Box::new(|_| Err(0)),
+            tokio_rt,
+            on_signal: Arc::new(|_| Err(0)),
```

Comment:
know it's not exactly the line within this PR, but @drunkirishcoder and I discussed him adding in an example within the comment doc for when a user wants to impl `set_on_signal`.